arbitrator/src/arbitrator_state_machine.cpp

```diff
@@ -27,13 +28,13 @@ namespace arbitrator
 
     ArbitratorState ArbitratorStateMachine::submit_event(ArbitratorEvent event)
     {
-        for (auto iter = ARBITRATOR_TRANSITIONS.begin(); iter != ARBITRATOR_TRANSITIONS.end(); iter++) 
+        for (auto pair : ARBITRATOR_TRANSITIONS) 
         {
-            if (current_state == iter->current_state && event == iter->input_event) {
+            if (current_state == pair.current_state && event == pair.input_event) 
+            {
                 // In the event of multiple legal transistions this will take the last transition,
                 // But ultimately this condition should never arise.
-                current_state = iter->final_state;
-                break;
```

Comment:
We still want to break out if this happens right? Why are we removing this?

Reply:
correct. this was a mistake.